File: ads.py
/**
 * Sistema de Anúncios para Jogos Unity WebGL
 * Versão: 1.0.0
 * 
 * Este script gerencia a exibição de anúncios em jogos Unity WebGL:
 * - Banners rotativos no topo da página
 * - Anúncios de tela cheia após game overs
 */

// Configuração do sistema de anúncios
const adSystem = {
    // URL da API do backend (atualizada para o Render)
    apiUrl: 'https://ads-system-backend.onrender.com',
    
    // Configurações dos anúncios
    config: {
        // Banner rotativo
        banner: {
            enabled: true,
            rotationInterval: 7000, // 7 segundos
            container: null,
            currentIndex: 0,
            ads: [],
            timer: null
        },
        
        // Anúncio de tela cheia
        fullscreen: {
            enabled: true,
            displayDuration: 5000, // 5 segundos
            gameOverThreshold: 5, // Exibir a cada 5 game overs
            container: null,
            ads: []
        }
    },
    
    // Contador de game overs
    gameOverCount: 0,
    
    /**
     * Inicializa o sistema de anúncios
     */
    init: function() {
        console.log('Sistema de anúncios inicializando...');
        
        // Criar containers para os anúncios
        this.createAdContainers();
        
        // Carregar anúncios do backend
        this.loadAds();
        
        // Adicionar listener para detectar quando o Unity estiver pronto
        window.addEventListener('unityInstance', (e) => {
            console.log('Unity instance detectada, sistema de anúncios pronto');
        });
        
        console.log('Sistema de anúncios inicializado');
    },
    
    /**
     * Cria os containers para os anúncios
     */
    createAdContainers: function() {
        // Container para banner
        const bannerContainer = document.createElement('div');
        bannerContainer.id = 'ad-banner-container';
        bannerContainer.style.position = 'absolute';
        bannerContainer.style.top = '0';
        bannerContainer.style.left = '0';
        bannerContainer.style.width = '100%';
        bannerContainer.style.height = '140px';
        bannerContainer.style.zIndex = '999';
        bannerContainer.style.display = 'none';
        document.body.appendChild(bannerContainer);
        this.config.banner.container = bannerContainer;
        
        // Container para anúncios de tela cheia
        const fullscreenContainer = document.createElement('div');
        fullscreenContainer.id = 'ad-fullscreen-container';
        fullscreenContainer.style.position = 'fixed';
        fullscreenContainer.style.top = '0';
        fullscreenContainer.style.left = '0';
        fullscreenContainer.style.width = '100%';
        fullscreenContainer.style.height = '100%';
        fullscreenContainer.style.backgroundColor = 'rgba(0, 0, 0, 0.8)';
        fullscreenContainer.style.zIndex = '1000';
        fullscreenContainer.style.display = 'none';
        fullscreenContainer.style.justifyContent = 'center';
        fullscreenContainer.style.alignItems = 'center';
        document.body.appendChild(fullscreenContainer);
        this.config.fullscreen.container = fullscreenContainer;
    },
    
    /**
     * Carrega os anúncios do backend
     */
    loadAds: function() {
        console.log('Carregando anúncios do backend...');
        
        // Carregar banners
        fetch(`${this.apiUrl}/api/banners`)
            .then(response => response.json())
            .then(data => {
                console.log('Banners carregados:', data.length);
                this.config.banner.ads = data;
                if (data.length > 0) {
                    this.startBannerRotation();
                }
            })
            .catch(error => {
                console.error('Erro ao carregar banners:', error);
            });
        
        // Carregar anúncios de tela cheia
        fetch(`${this.apiUrl}/api/fullscreen`)
            .then(response => response.json())
            .then(data => {
                console.log('Anúncios de tela cheia carregados:', data.length);
                this.config.fullscreen.ads = data;
            })
            .catch(error => {
                console.error('Erro ao carregar anúncios de tela cheia:', error);
            });
    },
    
    /**
     * Inicia a rotação de banners
     */
    startBannerRotation: function() {
        if (!this.config.banner.enabled || this.config.banner.ads.length === 0) {
            return;
        }
        
        // Exibir o primeiro banner
        this.showBanner(0);
        
        // Iniciar timer para rotação
        this.config.banner.timer = setInterval(() => {
            this.config.banner.currentIndex = (this.config.banner.currentIndex + 1) % this.config.banner.ads.length;
            this.showBanner(this.config.banner.currentIndex);
        }, this.config.banner.rotationInterval);
    },
    
    /**
     * Exibe um banner específico
     * @param {number} index - Índice do banner a ser exibido
     */
    showBanner: function(index) {
        const container = this.config.banner.container;
        const ad = this.config.banner.ads[index];
        
        if (!ad || !container) {
            return;
        }
        
        // Verificar se estamos na tela de jogo
        const isGameScreen = document.querySelector('.game-screen-indicator') && 
                            document.querySelector('.game-screen-indicator').style.display === 'block';
        
        if (!isGameScreen) {
            container.style.display = 'none';
            return;
        }
        
        // Limpar container
        container.innerHTML = '';
        
        // Criar elemento do banner
        const banner = document.createElement('div');
        banner.style.width = '1080px';
        banner.style.height = '140px';
        banner.style.maxWidth = '100%';
        banner.style.margin = '0 auto';
        banner.style.position = 'relative';
        banner.style.backgroundImage = `url(${ad.imageUrl})`;
        banner.style.backgroundSize = 'cover';
        banner.style.backgroundPosition = 'center';
        banner.style.cursor = 'pointer';
        
        // Adicionar evento de clique
        banner.addEventListener('click', () => {
            this.recordAdClick(ad.id, 'banner');
            window.open(ad.linkUrl, '_blank');
        });
        
        // Adicionar ao container
        container.appendChild(banner);
        container.style.display = 'block';
        
        // Registrar impressão
        this.recordAdImpression(ad.id, 'banner');
    },
    
    /**
     * Manipula evento de game over
     */
    handleGameOver: function() {
        this.gameOverCount++;
        console.log('Game over detectado, contagem:', this.gameOverCount);
        
        // Verificar se deve exibir anúncio de tela cheia
        if (this.config.fullscreen.enabled && 
            this.gameOverCount % this.config.fullscreen.gameOverThreshold === 0 &&
            this.config.fullscreen.ads.length > 0) {
            
            // Pausar o jogo
            if (typeof window.pauseGame === 'function') {
                window.pauseGame();
            }
            
            // Exibir anúncio de tela cheia
            this.showFullscreenAd();
        }
    },
    
    /**
     * Exibe um anúncio de tela cheia
     */
    showFullscreenAd: function() {
        const container = this.config.fullscreen.container;
        
        // Selecionar um anúncio aleatório
        const randomIndex = Math.floor(Math.random() * this.config.fullscreen.ads.length);
        const ad = this.config.fullscreen.ads[randomIndex];
        
        if (!ad || !container) {
            return;
        }
        
        // Limpar container
        container.innerHTML = '';
        
        // Criar elemento do anúncio
        const adElement = document.createElement('div');
        adElement.style.width = '1080px';
        adElement.style.height = '1920px';
        adElement.style.maxWidth = '90%';
        adElement.style.maxHeight = '90%';
        adElement.style.position = 'relative';
        adElement.style.backgroundImage = `url(${ad.imageUrl})`;
        adElement.style.backgroundSize = 'contain';
        adElement.style.backgroundPosition = 'center';
        adElement.style.backgroundRepeat = 'no-repeat';
        adElement.style.cursor = 'pointer';
        
        // Adicionar evento de clique
        adElement.addEventListener('click', () => {
            this.recordAdClick(ad.id, 'fullscreen');
            window.open(ad.linkUrl, '_blank');
        });
        
        // Adicionar botão de fechar
        const closeButton = document.createElement('div');
        closeButton.style.position = 'absolute';
        closeButton.style.top = '10px';
        closeButton.style.right = '10px';
        closeButton.style.width = '30px';
        closeButton.style.height = '30px';
        closeButton.style.borderRadius = '50%';
        closeButton.style.backgroundColor = 'rgba(0, 0, 0, 0.5)';
        closeButton.style.color = 'white';
        closeButton.style.textAlign = 'center';
        closeButton.style.lineHeight = '30px';
        closeButton.style.cursor = 'pointer';
        closeButton.innerHTML = '✕';
        closeButton.addEventListener('click', (e) => {
            e.stopPropagation();
            this.hideFullscreenAd();
        });
        
        adElement.appendChild(closeButton);
        
        // Adicionar ao container
        container.appendChild(adElement);
        container.style.display = 'flex';
        
        // Registrar impressão
        this.recordAdImpression(ad.id, 'fullscreen');
        
        // Configurar timer para fechar automaticamente
        setTimeout(() => {
            this.hideFullscreenAd();
        }, this.config.fullscreen.displayDuration);
    },
    
    /**
     * Esconde o anúncio de tela cheia
     */
    hideFullscreenAd: function() {
        const container = this.config.fullscreen.container;
        if (container) {
            container.style.display = 'none';
        }
        
        // Retomar o jogo
        if (typeof window.resumeGame === 'function') {
            window.resumeGame();
        }
    },
    
    /**
     * Registra uma impressão de anúncio
     * @param {string} adId - ID do anúncio
     * @param {string} adType - Tipo do anúncio ('banner' ou 'fullscreen')
     */
    recordAdImpression: function(adId, adType) {
        fetch(`${this.apiUrl}/api/impression`, {
            method: 'POST',
            headers: {
                'Content-Type': 'application/json'
            },
            body: JSON.stringify({
                adId: adId,
                adType: adType
            })
        })
        .catch(error => {
            console.error('Erro ao registrar impressão:', error);
        });
    },
    
    /**
     * Registra um clique em anúncio
     * @param {string} adId - ID do anúncio
     * @param {string} adType - Tipo do anúncio ('banner' ou 'fullscreen')
     */
    recordAdClick: function(adId, adType) {
        fetch(`${this.apiUrl}/api/click`, {
            method: 'POST',
            headers: {
                'Content-Type': 'application/json'
            },
            body: JSON.stringify({
                adId: adId,
                adType: adType
            })
        })
        .catch(error => {
            console.error('Erro ao registrar clique:', error);
        });
    }
};

// Inicializar o sistema de anúncios quando o DOM estiver pronto
document.addEventListener('DOMContentLoaded', function() {
    console.log('DOM carregado, inicializando sistema de anúncios...');
    adSystem.init();
});

// Notificar que o script foi carregado
console.log('Sistema de anúncios carregado ou tentativa de carregamento concluída');

File: ads.js
// ads_final.js (ou ads.js)

(function() { // Usamos uma IIFE para encapsular e evitar poluir o escopo global desnecessariamente
    // Verifica se a variável adSystem já existe para evitar redeclaração
    if (typeof window.adSystem !== 'undefined') {
        console.warn('window.adSystem já está definido. Pulando a redeclaração.');
        return; // Sai se já estiver definido
    }

    const adSystem = (function() {
        // Objeto para armazenar referências aos elementos do DOM dos anúncios
        let domElements = {
            bannerContainer: null,
            fullscreenContainer: null
        };

        // Estado interno do sistema de anúncios
        let systemState = {
            initialized: false,
            adsData: null, // Armazenará os dados dos anúncios carregados
            currentBannerIndex: 0,
            gameOverCounter: 0,
            isShowingFullscreen: false,
            bannerRotationTimer: null
        };

        // Configurações padrão
        const defaultConfig = {
            apiUrl: 'https://ads-system-backend.onrender.com',
            bannerRotationInterval: 7000, // 7 segundos
            fullscreenDuration: 5000,    // 5 segundos
            gameOverCountThreshold: 5,   // Mostrar fullscreen a cada 5 game overs
            debug: true,                 // Ativa logs de depuração
            retryAttempts: 3,            // Tentativas para buscar anúncios
            retryDelay: 3000,            // Atraso entre as tentativas
            fallbackImageUrl: 'https://via.placeholder.com/1080x140?text=Anuncio+Padrao',
            fallbackTargetUrl: 'https://alexlealdigital.github.io',
            containerZIndex: {
                banner: 2147483647, // Valor máximo garantido
                fullscreen: 2147483647
            }
        };

        let currentConfig = {}; // Configuração final após a inicialização

        /**
         * Log de depuração condicional.
         * @param {...any} args - Argumentos a serem logados.
         */
        function debugLog(...args) {
            if (currentConfig.debug) {
                console.log('[AdSystem Debug]', ...args);
            }
        }

        /**
         * Cria e anexa um elemento de contêiner de anúncio ao DOM.
         * @param {string} id - ID do contêiner.
         * @param {number} zIndex - Z-index do contêiner.
         * @returns {HTMLElement} O elemento do contêiner criado.
         */
        function createAdContainer(id, zIndex) {
            let container = document.getElementById(id);
            if (!container) {
                container = document.createElement('div');
                container.id = id;
                container.style.cssText = `
                    position: fixed;
                    left: 0;
                    width: 100%;
                    background-color: rgba(0, 0, 0, 0.8); /* Fundo semi-transparente para destaque */
                    color: white;
                    text-align: center;
                    display: none; /* Inicia oculto */
                    z-index: ${zIndex};
                    box-sizing: border-box; /* Garante que padding e border não aumentem o tamanho */
                `;
                document.body.appendChild(container);
            }
            return container;
        }

        /**
         * Inicializa os contêineres de banner e fullscreen.
         */
        function setupAdContainers() {
            domElements.bannerContainer = createAdContainer('ad-banner-container', currentConfig.containerZIndex.banner);
            domElements.bannerContainer.style.height = '140px'; // Altura padrão para banner
            domElements.bannerContainer.style.top = '0'; // Posição no topo

            domElements.fullscreenContainer = createAdContainer('ad-fullscreen-container', currentConfig.containerZIndex.fullscreen);
            domElements.fullscreenContainer.style.top = '0';
            domElements.fullscreenContainer.style.left = '0';
            domElements.fullscreenContainer.style.width = '100%';
            domElements.fullscreenContainer.style.height = '100%';
            domElements.fullscreenContainer.style.display = 'flex'; // Usar flexbox para centralizar
            domElements.fullscreenContainer.style.justifyContent = 'center';
            domElements.fullscreenContainer.style.alignItems = 'center';
            domElements.fullscreenContainer.style.backgroundColor = 'black'; // Fundo preto para fullscreen
        }

        /**
         * Ajusta o z-index do canvas do Unity para garantir que os anúncios fiquem por cima.
         */
        const safeUnityCanvasZIndex = () => {
            if (typeof window.unityInstance !== 'undefined' &&
                window.unityInstance &&
                window.unityInstance.Module &&
                window.unityInstance.Module.canvas) {
                debugLog('Ajustando z-index do canvas do Unity para 0.');
                window.unityInstance.Module.canvas.style.zIndex = '0';
            } else {
                debugLog('Unity instance ou canvas não encontrado para ajustar z-index.');
            }
        };

        /**
         * Busca os dados dos anúncios da API.
         * Implementa lógica de retry.
         * @param {number} attempt - Tentativa atual.
         * @returns {Promise<Array>} Promessa que resolve com os dados dos anúncios.
         */
        async function fetchAdsData(attempt = 1) {
            debugLog(`Tentando buscar dados de anúncios (Tentativa ${attempt}/${currentConfig.retryAttempts})...`);
            try {
                const response = await fetch(`${currentConfig.apiUrl}/ads`);
                if (!response.ok) {
                    throw new Error(`HTTP error! status: ${response.status}`);
                }
                const data = await response.json();
                systemState.adsData = data;
                debugLog('Dados de anúncios carregados:', data);
                return data;
            } catch (error) {
                console.error(`Erro ao buscar dados de anúncios na tentativa ${attempt}:`, error);
                if (attempt < currentConfig.retryAttempts) {
                    await new Promise(resolve => setTimeout(resolve, currentConfig.retryDelay));
                    return fetchAdsData(attempt + 1); // Tenta novamente
                } else {
                    console.error('Falha ao buscar dados de anúncios após todas as tentativas. Usando fallback.');
                    // Usar um anúncio de fallback se a API falhar
                    systemState.adsData = [{
                        type: 'banner',
                        imageUrl: currentConfig.fallbackImageUrl,
                        targetUrl: currentConfig.fallbackTargetUrl
                    }, {
                        type: 'fullscreen', // Pode ter um fallback fullscreen também
                        imageUrl: 'https://via.placeholder.com/1080x1920?text=Anuncio+Tela+Cheia+Padrao',
                        targetUrl: currentConfig.fallbackTargetUrl
                    }];
                    return systemState.adsData;
                }
            }
        }

        /**
         * Exibe um anúncio banner.
         */
        function showBanner() {
            if (!domElements.bannerContainer) {
                debugLog('Contêiner do banner não está pronto.');
                return;
            }

            if (!systemState.adsData || systemState.adsData.length === 0) {
                debugLog('Nenhum dado de anúncio disponível para banner. Exibindo fallback.');
                // Usar um banner de fallback se não houver dados
                const fallbackAd = {
                    imageUrl: currentConfig.fallbackImageUrl,
                    targetUrl: currentConfig.fallbackTargetUrl
                };
                displayAdInContainer(domElements.bannerContainer, fallbackAd, 'banner');
                return;
            }

            // Filtrar apenas anúncios do tipo banner
            const banners = systemState.adsData.filter(ad => ad.type === 'banner');

            if (banners.length === 0) {
                debugLog('Nenhum anúncio tipo banner encontrado. Exibindo fallback.');
                const fallbackAd = {
                    imageUrl: currentConfig.fallbackImageUrl,
                    targetUrl: currentConfig.fallbackTargetUrl
                };
                displayAdInContainer(domElements.bannerContainer, fallbackAd, 'banner');
                return;
            }

            // Exibir o banner atual e programar o próximo
            const adToShow = banners[systemState.currentBannerIndex];
            displayAdInContainer(domElements.bannerContainer, adToShow, 'banner');

            // Preparar para o próximo banner
            systemState.currentBannerIndex = (systemState.currentBannerIndex + 1) % banners.length;

            // Limpar timer anterior e iniciar novo para rotação
            if (systemState.bannerRotationTimer) {
                clearInterval(systemState.bannerRotationTimer);
            }
            systemState.bannerRotationTimer = setInterval(() => {
                debugLog('Rotacionando banner...');
                showBanner(); // Chama a si mesmo para rotacionar
            }, currentConfig.bannerRotationInterval);

            domElements.bannerContainer.style.display = 'block';
            debugLog('Banner visível.');
        }

        /**
         * Oculta o anúncio banner.
         */
        function hideBanner() {
            if (domElements.bannerContainer) {
                domElements.bannerContainer.style.display = 'none';
                if (systemState.bannerRotationTimer) {
                    clearInterval(systemState.bannerRotationTimer);
                    systemState.bannerRotationTimer = null;
                }
                debugLog('Banner oculto.');
            }
        }

        /**
         * Exibe um anúncio fullscreen.
         * @param {object} adData - Dados do anúncio a ser exibido.
         */
        async function showFullscreenAd(adData) {
            if (!domElements.fullscreenContainer || systemState.isShowingFullscreen) {
                debugLog('Contêiner fullscreen não está pronto ou já está exibindo um anúncio fullscreen.');
                return;
            }

            systemState.isShowingFullscreen = true;
            publicAPI.pauseGame(); // Pausa o jogo (chamada para o Unity)

            // Exibir o anúncio fullscreen
            await displayAdInContainer(domElements.fullscreenContainer, adData, 'fullscreen');
            domElements.fullscreenContainer.style.display = 'flex'; // Usar flex para centralização

            debugLog('Anúncio fullscreen visível. Ocultando em', currentConfig.fullscreenDuration / 1000, 'segundos.');

            // Ocultar após a duração e resumir o jogo
            setTimeout(() => {
                hideFullscreenAd();
                publicAPI.resumeGame(); // Resumir o jogo (chamada para o Unity)
            }, currentConfig.fullscreenDuration);
        }

        /**
         * Oculta o anúncio fullscreen.
         */
        function hideFullscreenAd() {
            if (domElements.fullscreenContainer) {
                domElements.fullscreenContainer.style.display = 'none';
                systemState.isShowingFullscreen = false;
                debugLog('Anúncio fullscreen oculto.');
            }
        }

        /**
         * Exibe um anúncio (banner ou fullscreen) em um contêiner específico.
         * @param {HTMLElement} container - O elemento DOM onde o anúncio será exibido.
         * @param {object} ad - Objeto de dados do anúncio.
         * @param {string} type - 'banner' ou 'fullscreen'.
         */
        function displayAdInContainer(container, ad, type) {
            container.innerHTML = ''; // Limpa o conteúdo anterior

            const link = document.createElement('a');
            link.href = ad.targetUrl || currentConfig.fallbackTargetUrl;
            link.target = '_blank'; // Abre em nova aba
            link.style.display = 'flex';
            link.style.justifyContent = 'center';
            link.style.alignItems = 'center';
            link.style.width = '100%';
            link.style.height = '100%';

            const img = document.createElement('img');
            img.src = ad.imageUrl || (type === 'banner' ? currentConfig.fallbackImageUrl : 'https://via.placeholder.com/1080x1920?text=Anuncio+Tela+Cheia+Padrao');
            img.style.maxWidth = '100%';
            img.style.maxHeight = '100%';
            img.style.objectFit = 'contain'; // Garante que a imagem se ajuste

            link.appendChild(img);
            container.appendChild(link);

            // Adicionar um botão de fechar para fullscreen (opcional, mas boa prática)
            if (type === 'fullscreen') {
                const closeButton = document.createElement('button');
                closeButton.textContent = 'X';
                closeButton.style.cssText = `
                    position: absolute;
                    top: 10px;
                    right: 10px;
                    background-color: rgba(255, 255, 255, 0.7);
                    border: none;
                    border-radius: 50%;
                    width: 30px;
                    height: 30px;
                    font-size: 18px;
                    cursor: pointer;
                    z-index: ${currentConfig.containerZIndex.fullscreen + 1}; /* Acima do anúncio */
                `;
                closeButton.onclick = (e) => {
                    e.stopPropagation(); // Evita que o clique vá para o link do anúncio
                    hideFullscreenAd();
                    publicAPI.resumeGame();
                };
                container.appendChild(closeButton);
            }
        }

        /**
         * Lida com o evento de Game Over do jogo.
         */
        function handleGameOver() {
            systemState.gameOverCounter++;
            debugLog('Game Over! Contador:', systemState.gameOverCounter);

            if (systemState.gameOverCounter >= currentConfig.gameOverCountThreshold) {
                systemState.gameOverCounter = 0; // Resetar contador
                const fullscreenAds = systemState.adsData.filter(ad => ad.type === 'fullscreen');
                if (fullscreenAds.length > 0) {
                    // Selecionar um anúncio fullscreen aleatoriamente
                    const adToShow = fullscreenAds[Math.floor(Math.random() * fullscreenAds.length)];
                    showFullscreenAd(adToShow);
                } else {
                    debugLog('Nenhum anúncio fullscreen disponível.');
                }
            }
        }

        const publicAPI = {
            /**
             * Inicializa o sistema de anúncios.
             * @param {object} customConfig - Configurações opcionais para sobrescrever as padrão.
             */
            init: async function(customConfig = {}) {
                if (systemState.initialized) {
                    debugLog('AdSystem já inicializado.');
                    return;
                }

                currentConfig = { ...defaultConfig, ...customConfig };
                debugLog('Inicializando AdSystem com configuração:', currentConfig);

                setupAdContainers(); // Cria os contêineres DOM
                safeUnityCanvasZIndex(); // Ajusta o z-index do canvas do Unity

                await fetchAdsData(); // Busca os dados dos anúncios

                systemState.initialized = true;
                debugLog('AdSystem inicialização completa.');

                // Dispara evento global para sinalizar que o sistema está pronto
                document.dispatchEvent(new CustomEvent('adSystemReady'));
            },

            showBanner: function() {
                if (!systemState.initialized) {
                    debugLog('AdSystem não inicializado, não pode mostrar banner.');
                    return;
                }
                showBanner();
            },

            hideBanner: function() {
                if (!systemState.initialized) {
                    debugLog('AdSystem não inicializado, não pode esconder banner.');
                    return;
                }
                hideBanner();
            },

            handleGameOver: function() {
                if (!systemState.initialized) {
                    debugLog('AdSystem não inicializado, não pode lidar com game over.');
                    return;
                }
                handleGameOver();
            },

            // Métodos que o AdSystem chama para o Unity (Unity chama setGameScreen, gameOver etc. e o AdSystem chama pauseGame/resumeGame)
            pauseGame: function() {
                debugLog('Chamando Unity para pausar o jogo...');
                if (window.unityInstance && typeof window.unityInstance.SendMessage === 'function') {
                    window.unityInstance.SendMessage('GameManager', 'PauseGame');
                } else {
                    console.warn('Unity instance não disponível para pausar o jogo.');
                }
            },

            resumeGame: function() {
                debugLog('Chamando Unity para resumir o jogo...');
                if (window.unityInstance && typeof window.unityInstance.SendMessage === 'function') {
                    window.unityInstance.SendMessage('GameManager', 'ResumeGame');
                } else {
                    console.warn('Unity instance não disponível para resumir o jogo.');
                }
            },

            // Para depuração:
            getSystemState: function() {
                return { ...systemState, config: currentConfig };
            }
        };

        return publicAPI;
    })();

    // Expõe o adSystem globalmente
    window.adSystem = adSystem;

    // Listener para o DOMContentLoaded para iniciar o AdSystem
    // Este script será injetado dinamicamente, então ele precisa saber quando o DOM está pronto.
    // Ou, se for carregado via <script defer> no head, o DOMContentLoaded já pode ter disparado.
    // O ideal é que este script seja carregado *antes* do script principal do Unity para garantir que window.adSystem esteja definido.
    // A inicialização (adSystem.init()) será disparada pelo `index.html` de forma robusta.

})();